go/interpreter/lfvm/instructions_test.go

```diff
@@ -1624,3 +1624,189 @@ func TestGeneralCall_ResultIsWrittenToStack(t *testing.T) {
+	one := *uint256.NewInt(1)
+	zero := *uint256.NewInt(0)
+	maxUint256 := *uint256.NewInt(0).Sub(uint256.NewInt(0), uint256.NewInt(1))
+
+	tests := map[OpCode]struct {
+		implementation func(*context) error
+		inputs         []inputData
+	}{
+		JUMP: {
+			implementation: opJump,
+			inputs: []inputData{
+				inputStack(maxUint256),
+			},
+		},
+		JUMPI: {
+			implementation: opJumpi,
+			inputs: []inputData{
+				inputStack(one, maxUint256),
+			},
+		},
+		MSTORE: {
+			implementation: opMstore,
+			inputs: []inputData{
+				inputStack(zero, maxUint256),
+			},
+		},
+		MSTORE8: {
+			implementation: opMstore8,
+			inputs: []inputData{
+				inputStack(zero, maxUint256),
+			},
+		},
+		MCOPY: {
+			implementation: opMcopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero),
+			},
+		},
+		MLOAD: {
+			implementation: opMload,
+			inputs: []inputData{
+				inputStack(maxUint256),
+			},
+		},
+		CALLDATACOPY: {
+			implementation: opCallDataCopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero),
+				inputStack(one, one, maxUint256),
+			},
+		},
+		RETURNDATACOPY: {
+			implementation: opReturnDataCopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero),
+				inputStack(zero, maxUint256, zero),
+				inputStack(one, zero, maxUint256),
+			},
+		},
+		CODECOPY: {
+			implementation: opCodeCopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero),
+			},
+		},
+		EXTCODECOPY: {
+			implementation: opExtCodeCopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero, zero),
+			},
+		},
+		SHA3: {
+			implementation: opSha3,
+			inputs: []inputData{
+				inputStack(maxUint256, zero),
+			},
+		},
+		// Note: Log0 tests stack values overflowing for all LOG operations
+		LOG0: {
+			implementation: func(c *context) error {
+				return opLog(c, 0)
+			},
+			inputs: []inputData{
+				inputStack(maxUint256, one),
+			},
+		},
+		// Note: call operations are quite complicated and they are to be tested
+		// on its own means. This is just a simple test to validate the test itself.
+		CALL: {
+			implementation: opCall,
+			inputs: []inputData{
+				inputStack(maxUint256, one, zero, zero, zero, zero, zero),
+				inputStack(zero, zero, maxUint256, one, zero, zero, zero),
+			},
+		},
+	}
+
+	for op, test := range tests {
+		t.Run(op.String(), func(t *testing.T) {
+			forEachRevision(t, op, func(t *testing.T, revision tosca.Revision) {
+
+				for _, input := range test.inputs {
+					ctxt := getEmptyContext()
+					ctxt.code = Code{{op, 0}}
+					ctxt.params.Revision = revision
+					ctxt.gas = 1 << 32
+
+					for _, v := range input {
+						ctxt.stack.push(&v)
+					}
+
+					err := test.implementation(&ctxt)
+					if want, got := errOverflow, err; want != got {
+						t.Fatalf("unexpected error, wanted %v, got %v", want, got)
+					}
+				}
+			})
+		})
+	}
+}
+
+func TestInstructions_CopyOpRoundStackValuesOnOverflow(t *testing.T) {
```

Comment:
If I understand correctly, this tests make sure that overflows in offsets are ignored if the size is zero. If so, this should be reflected in the name. I do not think it has anything to do with rounding ...